### ISSUE TITLE
cannot build due to private github package

### ISSUE THREAD

**Issue by ch1y1z1** [creeper12356/AltCampusLife#1]
installing dependency got error：

```
error An unexpected error occurred: "https://npm.pkg.github.com/download/@creeper12356/altcampuslifeservice/1.0.4/bd7e78c4b4a8b34f370a12eb3701748ea342efc3: Request failed \"401 Unauthorized\"".
```

**Comment by creeper12356** [comment]
Hello. Thanks for your issue!
I have changed the visibility of [altcampuslifeservice package](https://github.com/users/creeper12356/packages/npm/package/altcampuslifeservice) to **public** . You can try `npm i` again! 😺

**Comment by ch1y1z1** [comment]
Thanks for your reply!

Now we have visibility to your npm package, but still dont have read permission.

```
npm ERR! code E403
npm ERR! 403 403 Forbidden - GET https://npm.pkg.github.com/download/@creeper12356/altcampuslifeservice/1.0.4/bd7e78c4b4a8b34f370a12eb3701748ea342efc3 - Permission permission_denied: The token provided does not match expected scopes.
npm ERR! 403 In most cases, you or one of your dependencies are requesting
npm ERR! 403 a package version that is forbidden by your security policy, or
npm ERR! 403 on a server you do not have access to.
```

I'm not sure but it seems that
your npm package will [inherit the permissions of your repository](https://docs.github.com/en/packages/learn-github-packages/about-permissions-for-github-packages).
That's meaning we cannot download your package because we dont have read permission of your repository.

**Comment by creeper12356** [comment]
@ch1y1z1 
The first line of [Github Docs](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-npm-registry#authenticating-to-github-packages) reads: 
```
You need an access token to publish, install, and delete private, internal, and public packages.
```
That means, even a public github package needs **an access token** to install, which is inconvenient🥲. So I've changed the registry of **altcampuslifeservice** to [official npm registry](https://www.npmjs.com/package/@creeper12356/altcampuslifeservice) , you can try install from here. 🐱

--- 

I'm new to npm package publishing, so if you encounter other problems  when building this project, feel free to post it here. 🧡

**Comment by ch1y1z1** [comment]
@creeper12356 Thanks for your reply! It solves for me! 😸️

And I could finally build and deploy on my iPhone:

![IMG_7913](https://github.com/user-attachments/assets/509ff9a0-9f06-422f-bac2-4dae80164b0f)

But before that, there are still some problems, 

react-native seems to have [bug during iOS build in version 0.73, which has been fixed in 0.74](https://github.com/facebook/react-native/pull/42639#issuecomment-1952712743).
After upgrade react-native and some other dependencies to latest, I managed to build this app. But I'm not sure whether this         will encounter some breaking change or other problems.

**Comment by creeper12356** [comment]
@ch1y1z1 
Congratulations for you! 😸
I would consider upgrading react-native version to 0.74 when I'm free sometimes . You can try building and testing this app on iOS yourself first. 

---

By the way, I would like to close this issue since it has been solved. You can create other issues if encountering other problems. 😄
